code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1105,3 +1105,64 @@
 	id = "corazone"
 	description = "A medication used to treat pain, fever, and inflammation, along with heart attacks."
 	color = "#F5F5F5"
+
+/datum/reagent/medicine/modafinil
+	name = "Modafinil"
+	id = "modafinil"
+	description = "Long-lasting sleep suppressant that very slightly reduces stun and knockdown times. Overdosing has horrendous side effects and deals lethal oxygen damage, will knock you unconscious if not dealt with."
+	reagent_state = LIQUID
+	color = "#BEF7D8" // palish blue white
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	overdose_threshold = 10 // low overdose treshold due to very low and random metabolism rate
+	taste_description = "salt" // it actually does taste salty
+	var/overdose_progress = 0 // to track overdose progress
```

Comment:
Wait there's no general system for tiered overdoses?

Reply:
I'm as surprised as you are